### DISCUSSION TITLE
OpenID token - 401

### DISCUSSION THREAD

**Discussion by MikeKry** [OrchardCMS/OrchardCore#11476] [General]
Hi,

I am experiencing a problem with openid token validity, api returns 401 few minutes before expiration time. 

e.g. in OpenId_OpenIdTokenIndex, field ExpirationDate is 2022-04-01 09:56:30.000
but when I query api, I get 401 response 10 minutes before its expiration. 

I am storing token in singleton class and when its validity expires (datetime.utcnow + token.expires_in) it gets regenerated => after re-generate it returns 200 OK again.

Do you have any idea what can cause this? Is it possible that OC token is marked as expired sooner than it should ?

**Comment by Skrypt** [comment]
10 minutes is awkward because there's nothing that could cause this based on TimeZones or Daylight Saving Time issues. So, this 10 minutes delay comes from code that invalidates all before its time.

This https://github.com/OrchardCMS/OrchardCore/pull/11382 is one PR that got merged recently that could be the cause of the issue "I think".

Maybe you could file a proper issue with details of which version you are currently using.

**Reply by MikeKry** [reply]
I have noticed another guideline to this problem:
this is request to get token, notice the time that it was sent:
**2022-04-01 02:09:30.4606**|4.....|Start processing HTTP request POST "https://localhost:8000/connect/token"

in database there is this creation date => 2022-04-01 02:20:30.000
that is 10 minutes later than request was sent and received..
it is not always 10 minutes though, it varies, but creation date is never same as request time. 

But when I calculate time between request time and 401 response, it equals to one hour
(2022-04-01 02:09:30.4606 and 2022-04-01 03:09:30.4344)

So it seems that openid token inside has different validity than what is store in yessql db, cant it be something with datetimeoffset?

https://github.com/OrchardCMS/OrchardCore/blob/a452bc4524534691fa5e349ff12e81cbd2dcc359/src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Stores/OpenIdTokenStore.cs#L284

**Reply by Skrypt** [reply]
It could definitely be

**Reply by Skrypt** [reply]
Please open an issue

**Reply by MikeKry** [reply]
okay, thanks

https://github.com/OrchardCMS/OrchardCore/issues/11477

**Comment by MikeKry** [comment]
not a bug, explained in linked issue => misinterpretation of logs - serilog configuration has to use UTC timestamp.
